src/foam/java/refinements.js

```diff
@@ -1531,12 +1531,12 @@ foam.CLASS({
           p.fromCSVLabelMapping(map2);
         });
 
-        for ( String key : map2.keySet() ) {
-          map.put(getName() + "." + key, new foam.lib.csv.FromCSVSetter() {
+        for ( java.util.Map.Entry<String, foam.lib.csv.FromCSVSetter> entry : map2.entrySet() ) {
+          map.put(getName() + "." + entry.getKey(), new foam.lib.csv.FromCSVSetter() {
             public void set(foam.core.FObject obj, String str) {
               try {
                 if ( prop.get(obj) == null ) prop.set(obj, prop.of().newInstance());
-                map2.get(key).set((foam.core.FObject) prop.get(obj), str);
+                entry.getValue().set((foam.core.FObject) prop.get(obj), str);
               } catch ( Throwable t ) {
                 // ???
```

Comment:
It would be nice to log this 'debug' if possible.

Reply:
Oh wow, my eyes skipped right over that thing. I can see why someone put question marks there though - PropertyInfo isn't ContextAware, but it does have access to ClassInfo but that's not ContextAware either... maybe conditionally log if the class has a Property axiom `x` of type `Context`?

Reply:
Well think only a System.err makes sense, as this code will be called during log message creation.